### ISSUE TITLE
Python expression evaluation using @() not allowed as subexpression 

### ISSUE THREAD

**Issue by dwjbosman** [xonsh/xonsh#2640]
I am trying to do something like this:

```
def upload_keys(user, server):
    scp /home/@(user)/.ssh/* @(user)@@(server):.ssh/.
```
Unfortunately it seems that the @() operator needs to be surrounded with spaces. 
```
def upload_keys(user, server):
    scp /home/ @(user) /.ssh/* @(user) @ @(server) :.ssh/.
```
This however doesn't work, because now the scp commands sees separate arguments.

A work around could be this
```
def upload_keys(user, server):
    $USER=user
    $SERVER=server
    scp /home/$USER/.ssh/* $USER@$SERVER:.ssh/.
```
But this pollutes the environment and results in a lot of copying.

Any clean solution to work with Python expressions as part of shell commands?

**Comment by melund** [comment]
You can put the variables in a single @() ...

Something like this

snail@sea ~
$ user = 'snail'
. server ='sea'
. echo @(f'{user}@{server}:.ssh/.')
.
snail@sea:.ssh/.

**Comment by dwjbosman** [comment]
Thanks for you quick reply, 

Funny, I was just typing feedback trying this approach. Will continue using your suggestion.

**Comment by melund** [comment]
No problem. Feel free to close if you think this answers your question.

**Comment by scopatz** [comment]
Actually, I think this is a feature that would be nice to have.  I think it would involve extending `expand_path()` to handle the `@(expr)` syntax.

**Comment by dwjbosman** [comment]
Using @(expr) as part of a path/argurment would indeed be cleaner when compared with using the python string interpolation @(f'{user}@{server}:.ssh/.') approach.

I've looked at the expandvars function that expands ${x}. It uses a regular expression to find the ${x} tags. Using a regex for finding @{expr] with arbitrary python expressions would be very hard. Maybe its possible to to find @( and then use the xonsh Python parser?

**Comment by scopatz** [comment]
Ok! I was totally wrong about how to implement this, but it is now available at #2821
